library/src/main/java/com/raizlabs/android/dbflow/content/CursorLoader.java

```diff
@@ -0,0 +1,178 @@
+package com.raizlabs.android.dbflow.content;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.net.Uri;
+import android.support.v4.content.AsyncTaskLoader;
+
+import com.raizlabs.android.dbflow.config.FlowManager;
+import com.raizlabs.android.dbflow.runtime.FlowContentObserver;
+import com.raizlabs.android.dbflow.sql.ModelQueriable;
+import com.raizlabs.android.dbflow.structure.BaseModel;
+import com.raizlabs.android.dbflow.structure.Model;
+
+import java.io.FileDescriptor;
+import java.io.PrintWriter;
+import java.util.Arrays;
+
+/**
+ * Static library support version of the framework's {@link android.content.CursorLoader}. Used to
+ * write apps that run on platforms prior to Android 3.0.  When running on Android 3.0 or above,
+ * this implementation is still used; it does not try to switch to the framework's implementation.
+ * See the framework SDK documentation for a class overview.
+ */
+public class CursorLoader extends AsyncTaskLoader<Cursor> {
+    final ForceLoadContentObserver mObserver;
+
+    ModelQueriable<? extends Model> mModelQueriable;
+    Cursor mCursor;
+
+    private final class ForceLoadContentObserver extends FlowContentObserver {
+        public ForceLoadContentObserver() {
+            // super(new Handler());
+        }
+
+        @Override
+        public boolean deliverSelfNotifications() {
+            return true;
+        }
+
+        @Override
+        public void onChange(boolean selfChange) {
+            super.onChange(selfChange);
+            onContentChanged();
+        }
+
+        @Override
+        public void onChange(boolean selfChange, Uri uri) {
+            super.onChange(selfChange, uri);
+            onContentChanged();
+        }
+    }
+
+    /* Runs on a worker thread */
+    @Override
+    public Cursor loadInBackground() {
+        Cursor cursor = mModelQueriable.query();
+        if (cursor != null) {
+            // Ensure the cursor window is filled
+            cursor.getCount();
+        }
+        return cursor;
+    }
+
+    /* Runs on the UI thread */
+    @Override
+    public void deliverResult(Cursor cursor) {
+        if (isReset()) {
+            // An async query came in while the loader is stopped
+            if (cursor != null) {
+                cursor.close();
+            }
+            return;
+        }
+        Cursor oldCursor = mCursor;
+        mCursor = cursor;
+
+        if (isStarted()) {
+            super.deliverResult(cursor);
+        }
+
+        if (oldCursor != null && oldCursor != cursor && !oldCursor.isClosed()) {
+            oldCursor.close();
+        }
+    }
+
+    /**
+     * Creates an empty unspecified CursorLoader.  You must follow this with calls to {@link
+     * #setModelQueriable(ModelQueriable<? extends Model>)}}, etc to specify the query to perform.
+     */
+    public CursorLoader(Context context) {
```

Comment:
not instantiating ```mModelQueriable``` here is dangerous as it will crash on ```loadInBackground()```. We probably want to force devs to use the other constructor so remove this.